client/registry_login.go

```diff
@@ -0,0 +1,22 @@
+package client
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+// RegistryLogin authenticates the server with a given registry to login.
+func (client *APIClient) RegistryLogin(ctx context.Context, auth *types.AuthConfig) (*types.AuthResponse, error) {
+	resp, err := client.post(ctx, "/auth", nil, auth, nil)
+	if err != nil || resp.StatusCode == http.StatusUnauthorized {
```

Comment:
This part has a potential logic bug which is already described in https://github.com/alibaba/pouch/issues/1030.